akka/templates/deployment.yaml

```diff
@@ -0,0 +1,185 @@
+  strategy:
+    {{- toYaml .Values.strategy | nindent 4 }}
+  template:
+    metadata:
+      annotations:
+        {{- if .Values.metrics.enabled }}
+        prometheus.io/scrape: "true"
+        prometheus.io/port: "{{ .Values.metrics.port }}"
+        prometheus.io/path: "{{ .Values.metrics.path }}"
+        {{- end }}
+        {{- with .Values.podAnnotations }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+      labels:
+        {{- include "akka.labels" . | nindent 8 }}
+        {{- with .Values.podLabels }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+    spec:
+      affinity:
+        {{- toYaml .Values.affinity | nindent 8 }}
+      containers:
+        - args:
+            {{- if .Values.useResourceApplicationConf }}
+            - "-Dconfig.file=/opt/conf/chart.conf"
+            {{- end }}
+          command:
+            {{- toYaml .Values.command | nindent 12 }}
+          env:
+            {{- toYaml .Values.env | nindent 12 }}
+          envFrom:
+            {{- toYaml .Values.envFrom | nindent 12 }}
+          name: "{{ include "akka.containerName" . }}"
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tags }}"
+          imagePullPolicy: "{{ .Values.image.pullPolicy }}"
+          {{- if and .Values.akka.management.enabled .Values.livenessProbe.enabled }}
+          livenessProbe:
+            httpGet:
+              scheme: HTTP
+              path: /{{ .Values.akka.management.healthChecks.livenessPath }}
+              port: {{ .Values.akka.management.port }}
+            initialDelaySeconds: {{ .Values.livenessProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.livenessProbe.periodSeconds }}
+            timeoutSeconds: {{ .Values.livenessProbe.timeoutSeconds }}
+            successThreshold: {{ .Values.livenessProbe.successThreshold }}
+            failureThreshold: {{ .Values.livenessProbe.failureThreshold }}
+          {{- else if .Values.livenessProbe.enabled }}
+          livenessProbe:
+            {{- with .Values.livenessProbe.httpGet }}
+            httpGet:
+            {{- toYaml . | nindent 14 }}
+            {{- end }}
+            {{- with .Values.livenessProbe.tcpSocket }}
+            tcpSocket:
+            {{- toYaml . | nindent 14 }}
+            {{- end }}
+            initialDelaySeconds: {{ .Values.livenessProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.livenessProbe.periodSeconds }}
+            timeoutSeconds: {{ .Values.livenessProbe.timeoutSeconds }}
+            successThreshold: {{ .Values.livenessProbe.successThreshold }}
+            failureThreshold: {{ .Values.livenessProbe.failureThreshold }}
+          {{- end }}
+          lifecycle:
+            {{- toYaml .Values.lifecycle | nindent 12 }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          ports:
+            {{- if .Values.akka.http.enabled }}
+            - name: http
```

Comment:
I would be happy to change the name. Because one service opens multiple ports with http, so I can name each one. And it's easy to understand.

Reply:
@cw-sakamoto 
For multiple ports, you can specify the name in extraPorts.
Still want to rename a standard port?

Reply:
I made it changeable in `akka.http.portName`.